storage-node/src/server.rs

```diff
@@ -1,26 +1,105 @@
 use enum_as_inner::EnumAsInner;
+use std::fs;
+use std::path::Path;
+use storage_common::RequestParams;
+use tokio::fs::File as AsyncFile;
+use tokio_util::io::ReaderStream;
+use warp::{Filter, Rejection, Reply};
 
 use crate::{
     cache::lru::LruCache, disk::disk_manager_sync::DiskManagerSync, error::ParpulseResult,
     storage_manager::StorageManager,
 };
 
-// FIXME: Discuss with catalog team for more information.
-#[derive(Clone, EnumAsInner)]
-pub enum RequestParams {
-    File(String),
-    S3(String),
-}
-
 pub async fn storage_node_serve() -> ParpulseResult<()> {
     // TODO: Read the type of the cache from config.
     let dummy_size = 10;
     let cache = LruCache::new(dummy_size);
     let disk_manager = DiskManagerSync::default();
     let storage_manager = StorageManager::new(cache, disk_manager, "cache/".to_string());
 
-    // TODO:
-    // 1. Start the server here and listen on the requests.
-    // 2. Feed the request into the storage manager.
-    todo!()
+    // FIXME (kunle): We need to get the file from storage manager. For now we directly read
+    // the file from disk. I will update it after the storage manager provides the relevant API.
```

Comment:
Are we going to read the response from a local file or sth like `ResponseWriterStream`?

Reply:
Wdym by "read the response"? Right now, the server reads the local file as a stream.

Reply:
sth like
```rust
let stream_reader = s3_reader.into_stream();
while let Some(data) = stream_reader.next() {
    response_writer.write(data);
}
let response = response_writer.collect();
```
(just pseudo code)

Reply:
Are you talking about the disk/cache miss situation? In that case, I guess the server needs to read the response from `ResponseWriterStream`. If there is a disk/cache hit, then read from a local disk/cache file.

Reply:
> Are you talking about the disk/cache miss situation?

No. I'm talking about the general way that the storage manager returns data to the server. If we have sth like `ResponseWriterStream` then it shouldn't matter whether it is cache miss or cache hit. The server should receive the data from the storage manager in a unified way.

Reply:
Then it should be the stream.